### PR TITLE
bpo-29773: Add more cases for testing string to float conversion errors.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

File: Lib/test/test_float.py [hunk 1]
```diff
@@ -119,15 +119,29 @@ def test_float_memoryview(self):
...
+        # embedded NUL
+        check('123\x00')
+        check('123\x00 245')
+        check('123\x00245')
+        # byte string with embedded NUL
+        check(b'123\x00')
+        # non-UTF-8 byte string
+        check(b'123\xbd')
+        # lone surrogate in Unicode string
+        #check('123\udca0')
```
Comment (vstinner) [inline Lib/test/test_float.py]: I dislike commented test. Either fix it to catch UnicodeEncodeError (which would be a CPython implementation specific test), or remove it.

Maybe it's even worth it to enhance the error message for that one.

File: Lib/test/test_float.py [hunk 2]
```diff
@@ -119,15 +119,29 @@ def test_float_memoryview(self):
...
+            self.assertEqual(str(cm.exception),
+                'could not convert string to float: %r' % (s,))
+
+        check('\xbd')
+        check('123\xbd')
+        check('  123 456  ')
+        check(b'  123 456  ')
+
+        # non-ascii digits
+        check('\u0663\u0661\u0664!')
```
Comment (vstinner) [inline Lib/test/test_float.py]: I suggest '\u0663\u0661\u0664' + '!', at the first read, I understood that the error came from non-ASCII digits. Or explain that the error is on the invalid '!' character.

File: Lib/test/test_float.py [hunk 3]
```diff
@@ -119,15 +119,29 @@ def test_float_memoryview(self):
...
+        # embedded NUL
+        check('123\x00')
+        check('123\x00 245')
+        check('123\x00245')
+        # byte string with embedded NUL
+        check(b'123\x00')
+        # non-UTF-8 byte string
+        check(b'123\xbd')
```
Comment (vstinner) [inline Lib/test/test_float.py]: Maybe it's worth it to factorize tests which use the "same" text and bytes string  (text encoded to Latin1): all tests except of '\u0663\u0661\u0664!', no?

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Lib/test/test_float.py (hunk 3 above)
Comment (serhiy-storchaka) [reply Lib/test/test_float.py]: I prefer to be more explicit here.

**Review by vstinner — APPROVED** [review]

**Pull request merged**